CMakeLists.txt

```diff
@@ -262,6 +262,14 @@ set_target_properties(datachannel PROPERTIES
 	SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
 	CXX_STANDARD 17
 	CXX_VISIBILITY_PRESET default)
+
+if(APPLE)
+	set_target_properties(datachannel PROPERTIES
+		SOVERSION ${PROJECT_VERSION}
+		MACHO_COMPATIBILITY_VERSION 0
```

Comment:
Compatibility version should be at least 1.0.0, but setting a default value would also remove any possibility to prohibit loading the library after a breaking ABI change.

Reply:
Changed to `1.0.0`

Reply:
@PatTheMav Does this mean that `MACHO_COMPATIBILITY_VERSION` should now be manually increased to track ABI compatibility specifically for MacOS? I'd prefer not to have another version to update on each release, would it be possible to set it to something like `${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.0`? Otherwise, it is a bit issue to set a dummy value here in practice?

Reply:
I was under the impression that it could be set to `${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.0` or whichever other version number is used for ABI compatibility can be used. That's up to the project.

Reply:
> I was under the impression that it could be set to `${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.0` or whichever other version number is used for ABI compatibility can be used. That's up to the project.

`${PROJECT_VERSION_MAJOR}.0.0` is also possible if ABI compatibility is ensured within the same major version.

The requirements are outlined on the associated manage:

> Specifies the compatibility version number of the library.  When a library is loaded by dyld, the compatibility version is checked and if the program's version is greater that the library's version, it is an error.
>
> If the compatibility version number is not specified, it has a value of 0 and no checking is done when the library is used.

That's also why it should not default to `0.0.0`.

Reply:
Great, so `${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.0` sounds good to me. @Sean-Der does the value work?

Reply:
Alright I ran a few more tests on the change - the root issue is CMake creating 2 layers of symlinks, which is the main thing we attempt to fix:

* Setting `SOVERSION` and `VERSION` to the _same_ value avoids the 2 layers of symlinks.
* As such we should set _both_ to `${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}` (e.g. 0.18) which generates a single dynamic library file (`libdatachannel.0.18.dylib`) and a single symlink (`libdatachannel.dylib`)
* This ensures that when the minor version changes, the link requirement is not resolved anymore
* The MACHO headers can stay as is if the _full_ version strings are desired in the headers, otherwise they inherit the generic version parameters (albeit truncated to the minor version, so 0.18.0 in this example)